pkg/reloader/reloader.go

```diff
@@ -412,28 +482,87 @@ func hashFile(h hash.Hash, fn string) error {
 	return nil
 }
 
-func (r *Reloader) triggerReload(ctx context.Context) error {
-	req, err := http.NewRequest("POST", r.reloadURL.String(), nil)
+type PIDReloader struct {
+	pname string
+}
+
+func NewPIDReloader(pname string) *PIDReloader {
+	return &PIDReloader{
+		pname: pname,
+	}
+}
+
+func (pr *PIDReloader) TriggerReload(ctx context.Context) error {
+	procs, err := ps.Processes()
+	if err != nil {
+		return fmt.Errorf("list processes: %w", err)
+	}
+
+	var proc ps.Process
+	for i := range procs {
+		if pr.pname == procs[i].Executable() {
+			proc = procs[i]
+			break
+		}
+	}
+
+	if proc == nil {
+		return fmt.Errorf("failed to find process matching %q", pr.pname)
+	}
+
+	p, err := os.FindProcess(proc.Pid())
+	if err != nil {
+		return fmt.Errorf("find process err: %w", err)
+	}
+
+	if proc == nil {
```

Comment:
I think this should be `p`? We already check proc above.
```suggestion
	if p == nil {
```

Reply:
oops